src/paperqa/types.py

```diff
@@ -377,19 +378,18 @@ def populate_formatted_answers_and_bib_from_raw_answer(
         }
         name_bib = {}
 
-        # https://regex101.com/r/h2Ca20/1
-        for parenthetical in re.findall(r"\(([^)]*)\)", formatted_without_references):
-
+        for parenthetical in get_parenthetical_substrings(formatted_without_references):
             # now we replace eligible parentheticals with the deduped names
-            deduped_names = {
-                id_to_name_map.get(key, "") for key in get_citation_ids(parenthetical)
-            }
-
-            # replace the parenthetical with the deduped names
+            # while we preserve order and deduplicate
+            deduped_names: dict[str, None] = dict.fromkeys(
+                id_to_name_map.get(key)  # type: ignore[misc]
```

Comment:
```suggestion
                id_to_name_map[key]
```

Can't we do this, since we did `if id_to_name_map.get(key)` below

Reply:
yea will change